components/org.wso2.carbon.identity.application.authenticator.yahoo/src/main/java/org/wso2/carbon/identity/application/authenticator/yahoo/YahooOAuth2Authenticator.java

```diff
@@ -179,8 +139,14 @@ protected String getAuthenticateUser(AuthenticationContext context, Map<String,
     @Override
     protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
 
-        String userGUID = token.getParam(YahooOAuth2AuthenticatorConstants.USER_GUID);
-        return getUserInfoURL() + userGUID + YahooOAuth2AuthenticatorConstants.YAHOO_USER_DETAILS_JSON;
+        String userInfoURL = getAuthenticatorConfig().getParameterMap()
+                .get(YahooOAuth2AuthenticatorConstants.YAHOO_USERINFO_ENDPOINT);
```

Comment:
Why are we giving the precedence to the deprecated URL? Shouldn't it be the other way going forward

Reply:
This is not the deprecated URL. YahooOAuth2AuthenticatorConstants.YAHOO_USERINFO_ENDPOINT is used to read the URL defined in the application-authentication.xml [1]

[1] https://github.com/wso2/carbon-identity-framework/pull/3115/files#diff-d29a9eff7b14af6127ca080056a36441R132